libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1077,17 +1077,21 @@ void AP_InertialSensor::wait_for_sample(void)
         // accel and gyro. This normally completes immediately.
         bool gyro_available = false;
         bool accel_available = false;
-        while (!gyro_available || !accel_available) {
+        while (true) {
```

Comment:
Despite the fact that is a short code inside the loop, from a maintainance POV I really don't like finite loop statements depending on internal breaks. That's because commits are added, loops get longer, checks get complex and bad things happens. And when it happens that I have to do it, I do prefer the fancy version `for (;;)` :-) . Thus, even that I see that this code "runs better" now, I'm not sure that the gain is that relevant to assume the risk.

But that is just an opinion, don't take as a -1